test.injections_named_parameters & example.annotations: Fail compilation on gcc 9+

## Expected Behavior
`test.injections_named_parameters` and `example.annotations` should compile.

## Actual Behavior

gcc 9 may be released as early as tomorrow.  Using its current git build, or using the new gcc 10 git build, fails with this error:

````
[ 10%] Building CXX object extension/test/CMakeFiles/test.injections_named_parameters.dir/injections/named_parameters.cpp.o
cd /home/jamespharvey20/home/code/di.my.git/build/extension/test && /usr/bin/c++   -I/home/jamespharvey20/home/code/di.my.git/extension/test -I/home/jamespharvey20/home/code/di.my.git/include -I/
home/jamespharvey20/home/code/di.my.git/extension/include  -std=gnu++1y -fno-exceptions -pedantic -Wall -Wextra -Werror   -o CMakeFiles/test.injections_named_parameters.dir/injections
/named_parameters.cpp.o -c /home/jamespharvey20/home/code/di.my.git/extension/test/injections/named_parameters.cpp
In file included from /home/jamespharvey20/home/code/di.my.git/extension/test/injections/named_parameters.cpp:11:
/home/jamespharvey20/home/code/di.my.git/extension/include/boost/di/extension/injections/named_parameters.hpp:51:16: error: ISO C++ did not adopt string literal operator templates tak
ing an argument pack of characters [-Werror=pedantic]
   51 | constexpr auto operator""_s() {
      |                ^~~~~~~~
cc1plus: all warnings being treated as errors
````

(Same error for `example/annotations.cpp:60:16`.)

       60 | constexpr auto operator""_s() {
          |                ^~~~~~~~

## Steps to Reproduce the Problem

  1. Install gcc 9+
  1. `make test.injections_named_parameters` or `make example.annotations`

## Specifications

  - Version: git master (194dd9111)
  - Platform: x86-64
  - Subsystem: Arch Linux

--- a/example/annotations.cpp
+++ b/example/annotations.cpp
@@ -56,10 +56,17 @@ annotations2::annotations2(int i1, int i2, int i3) {
 template <char...>
 struct string {};
 
+#if (__GNUC__ >= 9)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic warning "-Wpedantic"
+#endif
 template <class T, T... Chars>
 constexpr auto operator""_s() {
   return string<Chars...>{};
 }
+#if (__GNUC__ >= 9)
+#pragma GCC diagnostic pop
+#endif
 #endif
 //->
 

--- a/extension/include/boost/di/extension/injections/named_parameters.hpp
+++ b/extension/include/boost/di/extension/injections/named_parameters.hpp
@@ -47,10 +47,17 @@ struct pair {
   long end{};
 };
 
+#if (__GNUC__ >= 9)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic warning "-Wpedantic"
+#endif
 template <class T, T... Chars>
 constexpr auto operator""_s() {
   return aux::integral_constant<long, const_hash(chars<Chars...>{}, sizeof...(Chars) + 1)>{};
 }
+#if (__GNUC__ >= 9)
+#pragma GCC diagnostic pop
+#endif
 
 long constexpr const_hash(char const* input, long m = 0, long i = 0) {
   return *input && i < m ? static_cast<long>(*input) + 33 * const_hash(input + 1, m, i + 1) : 5381;

